content/develop/connect/redis-for-vscode.md

```diff
@@ -0,0 +1,124 @@
+
+## Connect to Redis databases {#connect-db}
+
+Click on the Redis mark (the cursive **R**) in the VS Code menu to begin connecting a Redis database to VS Code.
+
+{{< image filename="images/dev/connect/vscode/vscode-initial.png" >}}
+
+Click on the **+ Connect database** button. A dialog will display in the main pane. In the image shown below, all the options have been checked to show the available details for each connection. These connection details are similar to those accessible from [`redis-cli`]({{< relref "/develop/connect/cli" >}}).
+
+{{< note >}}
+There is no way to change the logical database once you have selected it. If you need to connect to a different logical database, you need to add a separate database connection.
+{{< /note >}}
+
+{{< image filename="images/dev/connect/vscode/vscode-add-menu.png" >}}
+
+After filling out the necessary fields, click on the **Add Redis database** button. The pane on the left side, where you would normally see the Explorer view, shows your database connections.
+
+{{< image filename="images/dev/connect/vscode/vscode-cnx-view.png" >}}
+
+### Connection tools
+
+Several tools are displayed for each open connection.
+
+{{< image filename="images/dev/connect/vscode/vscode-cnx-tools.png" >}}
+
+Left to right, they are:
+
+- Refresh connection, which retrieves fresh data from the connected Redis database.
+- Edit connection, which shows a dialog similar to the one described in [Connect to Redis Databases](#connect-db) above.
+- Delete connection.
+- Open CLI. See [CLI tool](#cli) below for more information.
+- Sort keys, either ascending or descending.
+- Filter keys by key name or pattern, and by key type.
+- Add a new key by type: Hash, List, Set, Sorted Set, String, or JSON.
+
+## Key view
+
+Here's what you'll see when there are no keys in your database (the image on the left) and when keys are present (the image on the right).
+
+| View with no keys | View with JSON keys |
+|:---               |:---                     |
+| {{< image filename="images/dev/connect/vscode/vscode-no-keys.png" >}} | {{< image filename="images/dev/connect/vscode/vscode-w-keys.png" >}} |
+
+Redis for VS Code will automatically group the keys based on the one available setting, **Delimiter to separate namespaces**, which you can view by clicking on the gear icon in the top-right of the left side pane. Click on the current value to change it. The default setting is the colon (`:`) character.
+
+{{< image filename="images/dev/connect/vscode/vscode-settings.png" >}}
+
+Click on a key to display its contents.
+
+{{< image filename="images/dev/connect/vscode/vscode-key-view.png" >}}
+
+### Key editing tools
+
+There are several editing tools that you can use to edit key data. Each data type has its own editing capabilities. The following examples show edits to JSON data. Note that changes to keys are immediately written to the server.
```

Comment:
is it worth adding one more example with hashes?
Redis for VS Code supports hash field expiration, so this is a chance to promote Redis 7.4.

Reply:
also for hashes we can mention value formatters - a very useful feature (not available for JSONs)

Reply:
Will do. Look for next merge.